File: frontend/src/api/system/menu.ts
import request from '@/api/request'
import type { ApiResponse } from '@/types'

export interface MenuVO {
  id: number
  name: string
  permissionCode: string
  path: string
  component: string
  icon: string
  parentId: number
  sort: number
  type: number
  visible: boolean
  status: number
  children?: MenuVO[]
}

export function getMenus() {
  return request.get<ApiResponse<MenuVO[]>>('/menus')
}

export function getUserMenus() {
  return request.get<ApiResponse<MenuVO[]>>('/menus/user')
}

export function addMenu(data: any) {
  return request.post<ApiResponse>('/menus', data)
}

export function updateMenu(id: number, data: any) {
  return request.put<ApiResponse>(`/menus/${id}`, data)
}

export function deleteMenu(id: number) {
  return request.delete<ApiResponse>(`/menus/${id}`)
}

File: backend/src/main/java/com/dataagent/auth/entity/Menu.java
package com.dataagent.auth.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
@TableName("sys_menu")
public class Menu {
    @TableId(type = IdType.AUTO)
    private Long id;

    private String name;
    private String permissionCode;
    private String path;
    private String component;
    private String icon;
    private Long parentId;
    private Integer sort;
    private Integer type;
    private Integer visible;
    private Integer status;

    @TableField(exist = false)
    private List<Menu> children;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}
